### PR TITLE
[US-4104] Removal of EntityDetail classes

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| The concept is unclear and is misused many times
|New feature| No 
|BC breaks| Yes 
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Resources/views/Admin/Content/BestsellingProduct/list.html.twig [hunk 1]
```diff
@@ -9,32 +9,32 @@
     
     {{ render(controller('ShopsysFrameworkBundle:Admin/Domain:domainTabs')) }}
 
-    {% macro categoryTreeItem(categoryDetails, selectedDomainId, bestsellingProductsInCategories) %}
+    {% macro categoryTreeItem(categoriesWithPreloadedChildren, selectedDomainId, bestsellingProductsInCategories) %}
```
Comment (Miroslav-Stopka) [inline packages/framework/src/Resources/views/Admin/Content/BestsellingProduct/list.html.twig]: Future generations will thank you for this Rostislav. CategoryDetails vs CategoriesWithPreloadedChildren - the difference in the content description that results from the name of the variable is huge.

**Comment by Miroslav-Stopka** [comment]
I wrote already, I am satisfied Rostislav. Manipulation with the functionality provided by *Details is clearer and more predictable now. Predictability is something I very much appreciate. It brings me a feeling of sureness :) Thank you for this change

**Comment by Miroslav-Stopka** [comment]
The tests processed without a problem. The only curiosity is the number of executed queries when the category opens in the administration. Please check that.

File: packages/framework/src/Model/Product/Detail/ProductDetail.php [hunk 2]
```diff
@@ -1,125 +0,0 @@
...
-            $this->imagesById = $this->productDetailFactory->getImagesIndexedById($this->product);
-        }
-
-        return $this->imagesById;
-    }
-
-    /**
-     * @return bool
-     */
-    public function hasContentForDetailBox()
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply packages/framework/src/Model/Product/Detail/ProductDetail.php]: only in FE template for product detail (`Front/Content/detail.html.twig`) - it was used for detection whether render `<dl class="box-detail__info__params">` or not - that was solved by the css pseudoclass in my solution :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
Thank you @Miroslav-Stopka! :slightly_smiling_face: I checked the queries problem and it behaves the same on master so I will create separate issue for that :+1:

**Pull request merged**